app/controllers/case_workers/application_controller.rb

```diff
@@ -4,7 +4,7 @@ class CaseWorkers::ApplicationController < ApplicationController
   private
 
   def authenticate_case_worker!
-    unless user_signed_in? && current_user.case_worker?
+    unless user_signed_in? && current_user.rolable.is_a?(CaseWorker)
```

Comment:
I guess you might want to do current_user.rolable.case_worker?

although I think I do prefer current_user.persona.case_worker?